### PR TITLE
Fix proxy regex in venia-upward.yml

### LINKED ISSUE #1072: Minor regex issue in venia-upward.yml
The regex in [venia-upward.yml line 12](https://github.com/magento-research/pwa-studio/blob/develop/packages/venia-concept/venia-upward.yml#L12) has a little problem.

Current regex: `^/(graphql|rest|media|cache/)`

Problem is that it matches for example `/resting`, which I assume should not be the case.

Corrected regex (if slash after each keyword is required): `^/(graphql|rest|media|cache)/`

- `/rest/foo` match
- `/rest/` match
- `/rest` does **not** match
- `/resting` does **not** match

Suggested corrected regex (if `/rest` should also match): `^/(graphql|rest|media|cache)(/|$)`

- `/rest/foo` match
- `/rest/` match
- `/rest` also **match**
- `/resting` does **not** match

### PR DESCRIPTION
**Pull request by andreas-ateles**
## Description

Fix proxy regex in `venia-upward.yml`.

## Related Issue

Closes #1072 

## Motivation and Context

See #1072

## Verification Steps

- A request to `/resting` should no longer be proxied to Magento and should not show the full Magento 404 page, but instead show an UPWARD 404 error.
- `/rest` should still show the proxied response.

## How Has This Been Tested?

The updated regex has been tested against test strings as described in #1072

## Proposed Labels for Change Type/Package

Bug fix in venia-concept.

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have linked an issue to this PR.
- [x] I have indicated the change type and relevant package(s).
- [x] I have proposed a change level by adding one of the `version: Major`, `version: Minor`, or `version: Patch` labels based on the defined [Public API](https://magento-research.github.io/pwa-studio/technologies/versioning/).
- [x] I have updated the documentation accordingly, if necessary.
- [x] I have added tests to cover my changes, if necessary.
- [ ] All new and existing tests passed.
- [ ] All CI checks are green (linting, build/deploy, etc).
- [ ] At least one core contributor has approved this PR.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento-research/pwa-studio?pullRequest=1073) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-research/pwa-studio?pullRequest=1073) before we can accept your contribution.<br/>

**Comment by andreas-ateles** [comment]
Not sure how I can apply a label here, but it should be: `version: Patch`

**Comment by zetlen** [comment]
Good catch! This was definitely unintended behavior and we appreciate the fix.

File: packages/venia-concept/venia-upward.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ response:
       pattern: '^/img/resize/(\d+)'
       use: imageOpt
     - matches: request.url.pathname
-      pattern: '^/(graphql|rest|media|cache/)'
+      pattern: '^/(graphql|rest|media|cache)(/|$)'
```
[the remark this answers is not in the document]
Comment (andreas-ateles) [reply packages/venia-concept/venia-upward.yml]: Sure. Fixed and rebase it on top of latest develop.

**Comment by andreas-ateles** [comment]
Removed the `cache` path and rebased on top of current develop.

**Pull request merged**